client/src/reducers/world.js

```diff
@@ -246,21 +261,25 @@ const WorldReducer = (
     case "annotation: label current cell selection": {
       const { metadataField, label } = action;
       const { crossfilter } = prevSharedState;
-      const mask = crossfilter.allSelectedMask();
-      const unclipped = {
-        ...state.unclipped,
-        obsAnnotations: AnnotationsHelpers.setLabelByMask(
-          state.unclipped.obsAnnotations,
-          metadataField,
-          mask,
-          label
-        )
+      return {
+        ...state,
+        ...setLabelOnCurrentSelection(state, crossfilter, metadataField, label)
       };
-      const obsAnnotations = state.obsAnnotations.replaceColData(
-        metadataField,
-        unclipped.obsAnnotations.col(metadataField).asArray()
-      );
-      return { ...state, obsAnnotations, unclipped };
+      // const mask = crossfilter.allSelectedMask();
```

Comment:
Oops, I think I left this dead code.  Please remove the commented section as it is detritus.